### ISSUE TITLE
Add support for using the library with mypy

### ISSUE THREAD

**Issue by ZeeD** [acederberg/pydantic-settings-yaml#18]
At the moment if I import your module I have
```
error: Skipping analyzing "yaml_settings_pydantic": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

**Comment by acederberg** [comment]
Hi @ZeeD! 

Thank you for using this library. I've reproduced this error in `nvim`. I'll take care of this. I'm currently occupied but I can get it done. It looks like I just need to add this

```toml
[tool.setuptools.package-data]
"pkgname" = ["py.typed"]
```

to `pyproject.toml` and add `py.typed` to the source folder and it should be good to go.

**Comment by acederberg** [comment]
I added the above and installed the package in editable mode and it works in `nvim`:

![image](https://github.com/acederberg/pydantic-settings-yaml/assets/77076023/23004453-9dae-4793-afda-291c275e0b15)

Changes are on `fix/py-typed`. Let me know that it works for you and I'll merge and release it.

**Comment by ZeeD** [comment]
Hi @acederberg !
thanks for the quick response and the changes!
yeah, it should be good with just the `py.typed` in the published package

**Comment by qs5779** [comment]
I am not sure why the fix/py-typed branch was never merged, but I just submitting a similar [PR](https://github.com/acederberg/pydantic-settings-yaml/pull/28)

**Comment by acederberg** [comment]
I thought I had fixed this, sorry @ZeeD! It is now merged in tag ``2.3.1``.
